libs/ledger/src/chain/main_chain.cpp

```diff
@@ -806,11 +813,20 @@ void MainChain::RecoverFromFile(Mode mode)
 
     if (in.is_open())
     {
-      byte_array::ByteArray bloom_filter_data{in};
+      try
+      {
+        byte_array::ByteArray bloom_filter_data{in};
 
-      LargeObjectSerializeHelper buffer{bloom_filter_data};
+        LargeObjectSerializeHelper buffer{bloom_filter_data};
 
-      buffer >> bloom_filter_;
+        buffer >> bloom_filter_;
+      }
+      catch (std::exception const &e)
+      {
+        FETCH_LOG_ERROR(LOGGING_NAME,
+                        "Failed to load Bloom filter from storage! Reason: ", e.what());
+        Reset();
```

Comment:
What happens with this in practise? Is there any easy point in the code that we can signal in the block coordinator that we need to restart from genesis again?

Reply:
Added the signal to block coordinator to reset

Reply:
my testing has found that this happens every time - so this is not a fix?